File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int mini=prices[0];
        int profit=0;
        if(prices.length==1){
            return 0;
        }
        for(int i=1;i<prices.length;i++){
            if(prices[i]<mini){
                mini=prices[i];
            }
            profit=Math.max(profit,prices[i]-mini);
        }
        return profit;

    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        minnum=prices[0]
        maxprofit=0

        for i , num in enumerate(prices):
            minnum=min(minnum,prices[i])
            maxprofit=max(maxprofit , (prices[i]-minnum))

        return maxprofit

        